ethstats/ethstats.go

```diff
@@ -183,6 +204,75 @@ func (s *Service) loop() {
 	}
 }
 
+// readLoop loops as long as the connection is alive and retrieves data packets
+// from the network socket. If any of them match an active request, it forwards
+// it, if they themselves are requests it initiates a reply, and lastly it drops
+// unknown packets.
+func (s *Service) readLoop(conn *websocket.Conn, in *json.Decoder) {
+	// If the read loop exists, close the connection
+	defer conn.Close()
```

Comment:
Generally libraries strive to allow mutliple closes particularly for these kinds of scenarios where multiple error locations should trigger the same teardown.

I tried running 10 close operations one after the other immediately after establishing connection, it didn't seem to bother it, no weird behavior. I also tester scenarios if the read loop dies (which always double closes as the main loop will also close) but no strange behavior was seen.